mautic/mautic
# Salesforce push leads contacts

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR, now checks if a contact exists with a matching email address if pushing from Mautic to Salesforce, this is to prevent duplicating email addresses. If a contact exists with that email address then priority is given to the contact. Otherwise a lead is created in salesforce
[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. Apply PR and push contacts to integration through a campaign or form action or through the pushleads command in Salesforce. See that if a contact exists with a matching email address in Mautic, then that will be created, whereas before a lead would be created even if the contact existed.
